tests/helpers/datasets.py

```diff
@@ -0,0 +1,187 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import os
+import random
+import time
+import urllib.request
+from typing import Optional, Sequence, Tuple
+
+import torch
+from torch import Tensor
+from torch.utils.data import Dataset
+
+
+class MNIST(Dataset):
+    """
+    Customized `MNIST <http://yann.lecun.com/exdb/mnist/>`_ dataset for testing Pytorch Lightning
+    without the torchvision dependency.
+    Part of the code was copied from
+    https://github.com/pytorch/vision/blob/build/v0.5.0/torchvision/datasets/mnist.py
+    Args:
+        root: Root directory of dataset where ``MNIST/processed/training.pt``
+            and  ``MNIST/processed/test.pt`` exist.
+        train: If ``True``, creates dataset from ``training.pt``,
+            otherwise from ``test.pt``.
+        normalize: mean and std deviation of the MNIST dataset.
+        download: If true, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again.
+    Examples:
+        >>> dataset = MNIST(".", download=True)
+        >>> len(dataset)
+        60000
+        >>> torch.bincount(dataset.targets)
+        tensor([5923, 6742, 5958, 6131, 5842, 5421, 5918, 6265, 5851, 5949])
+    """
+
+    RESOURCES = (
```

Comment:
The files are not that large, maybe let's just commit them to the repo? Especially considering we actually use 1000 images, so the total size will be about 1M if we "inline" the code that generates TrialMNIST. Having test data inlined greatly simplifies testing for cases where test machines don't have internet access (good for security).

We can keep the script the downloads and generates the data here though for reproducibility

Reply:
@Borda what do you think?

Reply:
well if it is just for testing we can make them as LFS to reduce repo size...

Reply:
I think this will be pretty small file - under 1 MB IIRC (if we only take TrialMNIST and save preprocessed version that we are actually using) which might not justify LFS, putting it into repo might be just fine?

Reply:
yes, but then why not have a synthetic dataset rather?

Reply:
@SkafteNicki ^^ ?

Reply:
remove the mnist to just run the test on synthetic dataset as proposed